### ISSUE TITLE
Don't delete persistence volume

### ISSUE THREAD

**Issue by jasperkuperus** [traefik/traefik-helm-chart#118]
First off: Great work on the latest changes. I could throw away a few home-baked things in favour of your updates. 😊 

The new `persistence` feature looks really nice. I tried it out and it works. But when I delete my Helm release, it also deletes this volume. This volume contains SSL certificates that are important and should not be thrown away that easily. I would prefer if the volume would stay intact, so you could easily do a `helm delete ...` and then `helm install ...` without losing precious SSL certificates.

**Comment by dtomcej** [comment]
Good catch.

We should add the helm resource policy for that:

https://github.com/helm/helm/issues/1875

https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource

however, if helm creates it, and doesn't delete it, then it becomes orphaned, and may cause issues with the second `helm install`.

@jasperkuperus Is there a situation that you would delete and then re-install? or is allowing your PVC to remain after the delete so that you can back it up etc would be sufficient?

**Comment by jasperkuperus** [comment]
I've had situations where I'd do a `helm delete` to get a cluster down quickly, or when helm wasn't able to upgrade the cluster. Then a clean slate with a clean install is the solution for me sometimes. It's not so much for backups, I just want the physical disk to stay in place, so I don't loose my certificates.

I have currently solved it by creating my own PVC:

```
{{ if .Values.global.tls }}
apiVersion: v1
kind: PersistentVolume
metadata:
  name: {{ .Release.Name }}-lets-encrypt-cert-volume
spec:
  storageClassName: ""
  capacity:
    storage: 10G
  accessModes:
    - ReadWriteMany
  gcePersistentDisk:
    pdName: {{ .Release.Name }}-lets-encrypt-cert-volume
    fsType: ext4
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: {{ .Release.Name }}-lets-encrypt-cert-volume-claim
spec:
  # It's necessary to specify "" as the storageClassName
  # so that the default storage class won't be used, see
  # https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
  storageClassName: ""
  volumeName: {{ .Release.Name }}-lets-encrypt-cert-volume
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 10G
{{ end }}
```

Removing the kubernetes volume/pvc is not a problem for me. But the actual disk for this volume/pvc is.

**Comment by jasperkuperus** [comment]
I tried the resource policy btw, which did not help me. The solution as described in my previous comment keeps the disk alive, but makes it possible to remove the pvc/volume. I see there's some GCP specific stuff in there. Not sure if we could do without.

**Comment by jakubhajek** [comment]
Hello @jasperkuperus 

Thanks for pointing out that. It is already implemented in #389 by using the specific resource-policy `helm.sh/resource-policy: keep`. Saying that in a simple word - uninstalling the chart will not remove the Traefik PVC.

**Comment by jasperkuperus** [comment]
> Hello @jasperkuperus
> 
> Thanks for pointing out that. It is already implemented in #389 by using the specific resource-policy `helm.sh/resource-policy: keep`. Saying that in a simple word - uninstalling the chart will not remove the Traefik PVC.

That's great! I'm not really in a position to test this out any time soon though. But if that PR covers it, I guess we can close this issue for now.
